internal/protocol/s3.go

```diff
@@ -120,3 +123,55 @@ func (S3BucketInfoTranslator) Validate(
 	}
 	return nil
 }
+
+func (S3CredentialTranslator) RpcToApi(c *cosiproto.S3CredentialInfo) map[cosiapi.CredentialVar]string {
+	if c == nil {
+		return nil
+	}
+
+	out := map[cosiapi.CredentialVar]string{
+		cosiapi.CredentialVar_S3_AccessKeyId:     c.AccessKeyId,
+		cosiapi.CredentialVar_S3_AccessSecretKey: c.AccessSecretKey,
+	}
+
+	return out
+}
+
+func (S3CredentialTranslator) ApiToRpc(vars map[cosiapi.CredentialVar]string) *cosiproto.S3CredentialInfo {
+	if len(vars) == 0 {
+		return nil
+	}
+
+	out := &cosiproto.S3CredentialInfo{}
+
+	out.AccessKeyId = vars[cosiapi.CredentialVar_S3_AccessKeyId]
+	out.AccessSecretKey = vars[cosiapi.CredentialVar_S3_AccessSecretKey]
+
+	return out
+}
+
+func (S3CredentialTranslator) Validate(
+	vars map[cosiapi.CredentialVar]string, authType cosiapi.BucketAccessAuthenticationType,
+) error {
+	// credentials are only required when authentication type is "Key"
+	if authType != cosiapi.BucketAccessAuthenticationTypeKey {
+		return nil
+	}
```

Comment:
same here, I didn't find any "ServiceAccount" auth type related specifications in `protocols.go`... so only added case for "Key" type for now...